anomalib/post_processing/visualizer.py

```diff
@@ -14,36 +14,193 @@
 # See the License for the specific language governing permissions
 # and limitations under the License.
 
+from dataclasses import dataclass, field
 from pathlib import Path
-from typing import Dict, List, Optional
+from typing import Dict, Iterator, List, Optional
 
 import cv2
 import matplotlib.figure
 import matplotlib.pyplot as plt
 import numpy as np
+from skimage.segmentation import mark_boundaries
+
+from anomalib.post_processing import (
+    add_anomalous_label,
+    add_normal_label,
+    superimpose_anomaly_map,
+)
+from anomalib.pre_processing.transforms import Denormalize
+
+
+@dataclass
+class ImageResult:
+    """Collection of data needed to visualize the predictions for an image."""
+
+    image: np.ndarray
+    pred_score: float
+    pred_label: str
+    anomaly_map: np.ndarray
+    gt_mask: Optional[np.ndarray] = None
+    pred_mask: Optional[np.ndarray] = None
+
+    heat_map: np.ndarray = field(init=False)
+    segmentations: np.ndarray = field(init=False)
+
+    def __post_init__(self):
+        """Generate heatmap overlay and segmentations, convert masks to images."""
+        self.heat_map = superimpose_anomaly_map(self.anomaly_map, self.image, normalize=False)
+        if self.pred_mask is not None and not np.max(self.pred_mask) == 255:
```

Comment:
I'll change the check to `max(pred_mask) <= 1`. In any case it is unlikely that the image is in the [0, 255] range, because the predicted mask is the result of the thresholding mechanism applied to the anomaly maps.

Reply:
sure, that makes sense. I was just thinking that in case it's used as an API inside or outside anomalib.